src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java

```diff
@@ -68,6 +71,38 @@ public CodeQualityServiceImpl(CodeQualityRepository codeQualityRepository,
         return searchType(request);
     }
 
+    @Override
+    public Iterable<CodeQuality> getAllSecurityScans(CodeQualityRequest request) {
+
+        Component component = componentRepository.findOne(request.getComponentId());
+
+        Iterable<CollectorItem> collectorItems = component.getCollectorItems(request.getType().collectorType());
+
+        ArrayList<CodeQuality> codeQualities = new ArrayList<CodeQuality>();
+
+        collectorItems.forEach(item -> {
```

Comment:
can this transformation happen on the client side?

Reply:
Discussed with Ragha, decided to leave it here and change the endpoint and function names to indicate it's only for UI use